src/storage/src/source/mysql/snapshot.rs

```diff
@@ -524,11 +526,20 @@ where
 #[must_use]
 fn build_snapshot_query(outputs: &[SourceOutputInfo]) -> String {
     let info = outputs.first().expect("MySQL table info");
+    if outputs.len() > 1
+        && !outputs
+            .iter()
+            .skip(1)
+            .map(|other_info| &other_info.desc.columns)
+            .all(|other_columns| *other_columns == info.desc.columns)
```

Comment:
small nit: in idiomatic Rust, at least in the MZ codebase, it usually better to avoid long chains of iterator transformations with either a short chain if possible or just a imperative for loop. This functional style becomes harder to read after a certain point. 

So I would write this either as:

 ```rust
assert!(outputs.iter().all(|o| info.desc.columns == o.desc.columns), "Mismatch...");
 ```
 
 or:
 
 ```rust
for output in outputs[1..] {
    assert!(info.desc.columns == output.desc.columns), "Mismatch...");
}
```